src/main/java/org/veupathdb/service/eda/ss/service/MetadataCache.java

```diff
@@ -115,17 +115,29 @@ private void invalidateOutOfDateStudies() {
     List<Study> studiesToRemove = _studies.values().stream()
         .filter(study -> isOutOfDate(study, dbStudies))
         .toList();
+
     synchronized (this) {
       LOG.info("Removing the following out of date or missing studies from cache: "
           + studiesToRemove.stream().map(StudyOverview::getStudyId).collect(Collectors.joining(",")));
 
-      // For each study with a study overview, check if the files exist and cache the result.
-      dbStudies.forEach(study -> _studyHasFilesCache.put(study.getStudyId(), _binaryFilesManager.studyHasFiles(study.getStudyId())));
+      dbStudies.forEach(study -> {
+        boolean studyHasFiles = _binaryFilesManager.studyHasFiles(study.getStudyId());
+
+        // Check if the files exist and cache the result. Store previous value in cache.
```

Comment:
Maybe move this up a line and modify
```suggestion
        // Check if the files exist and cache the result. Get previous value from cache at the same time.
```